docs(helmet): deprecate stencil helmet (#938)
ionic-team/stencil-site: src/docs/static-site-generation/meta.md

contributors:
  - mlynch
  - adamdbradley
  - rwaskiewicz
---

# SEO Meta Tags and Static Site Generation

Web Apps need to list detailed meta information about content in order to maximize SEO and provide good social media embed experiences.

One of the benefits to Stencil's prerendering is that most DOM apis are available in the NodeJS environment too.
For example, to set the document title, simply run `document.title = "Page Title"`.

> Or meta tags can be added and updated no differently than using the usual DOM APIs found in the browser, such as `document.head` and `document.createElement('meta')`.

Reviewer: Possible rewording section (I know this was already here, but figure why not tweak a little if it's kinda odd)

```suggestion
Similarly, meta tags can be set using standard DOM APIs as found in the browser, such as `document.head` and `document.createElement('meta')`.
```

> This mean your components runtime may already be able to do much of the custom work throughout prerendering.

Reviewer: ```suggestion
For this reason, your component's runtime can take care of much of this custom work during prerendering.
```

> That said, the Prerender Config also comes with many options that allows individual pages to be modified.

Reviewer: ```suggestion
That said, the Prerender Config also includes options that allow individual pages to be modified arbitrarily during prerendering.
```

> For example, the `afterHydrate(document, url)` hook can be used to update the parsed `document`, before it is serialized into an HTML string.

Reviewer: ```suggestion
For example, the `afterHydrate(document, url)` hook can be used to update the parsed `document` before it is serialized into an HTML string.
```

> The `document` argument can be used no different from the `document` found in a webpage, and the `url` argument is a `URL` location of the page being rendered.

Reviewer: ```suggestion
The `document` argument is a [standard `Document`](https://developer.mozilla.org/en-US/docs/Web/API/Document), while the `url` argument is a [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) for the location of the page being rendered.
```
